packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWRelatedCommunityCard/CWRelatedCommunityCard.tsx

```diff
@@ -1,112 +1,202 @@
+    if (!isCommunityMember && !hasJoinedCommunity) {
+      const joined = await handleJoinCommunity();
+
+      if (joined) {
+        setModeOfManageCommunityStakeModal('buy');
+      }
+    } else {
+      setModeOfManageCommunityStakeModal('buy');
+    }
+  };
+
   return (
-    <div
-      className={ComponentType.RelatedCommunityCard}
-      onClick={(e) => handleClick(e, id)}
-    >
-      <div className="content-container">
-        <div className="top-content">
-          <div className="header">
-            <CWCommunityAvatar community={communityAvatar} size="large" />
-            <CWText type="h5" title={communityName} fontWeight="medium">
-              {communityName}
-            </CWText>
-          </div>
-          <div className="description">
-            {communityDescription
-              ? addPeriodToText(communityDescription)
-              : null}
-          </div>
-        </div>
-        <div className="metadata">
-          <div className="member-data">
-            <CWIcon iconName="users" iconSize="small" />
-            <span className="count">
-              {Number(memberCount).toLocaleString('en-US')}
-            </span>
+    <>
+      <div
+        className={ComponentType.RelatedCommunityCard}
+        onClick={(e) => handleCommunityClick(e)}
+      >
+        <div className="content-container">
+          <div className="top-content">
+            <div className="community-info">
+              <div className="header">
+                <div className="community-name">
+                  <CWCommunityAvatar
+                    community={community as ChainInfo}
+                    size="large"
+                  />
+                  <CWText type="h5" title={community.name} fontWeight="medium">
+                    {community.name}
+                  </CWText>
+                </div>
 
-            <span className="text">
-              {pluralizeWithoutNumberPrefix(Number(memberCount), 'member')}
-            </span>
+                {!isNaN(stakeValue) && stakeChange && (
+                  <div className="stake-info">
+                    <CWText type="h5" className="stake-value">
+                      ${stakeValue}
+                    </CWText>
+                    <div>
+                      <CWText type="caption" className="stake-change">
+                        <span
+                          className={`percentage ${
+                            stakeChange >= 0 ? 'positive' : 'negative'
+                          }`}
+                        >
+                          {stakeChange}%
+                        </span>
+                        <span className="hours">24h</span>
+                      </CWText>
+                    </div>
+                  </div>
+                )}
+              </div>
+              <div className="description">
+                {community.description
+                  ? addPeriodToText(community.description)
+                  : null}
+              </div>
```

Comment:
use CWText

Reply:
actually in many other places in this PR, instead of using p, span, div, use CWText